### PR TITLE
Fix uploading artifacts even when the tests are successful

### PR DESCRIPTION
**Pull request by kevin940726**
## What?

Fix uploading artifacts even when Playwright tests are successful.

## Why?

Saves some bytes.

## How?

Only upload the `test-results` folder and check if it exists.

## Testing Instructions

CI should pass and no failure artifacts should be uploaded.

### CODE REVIEW

File: .github/workflows/end2end-test-playwright.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
               if: always()
               with:
                   name: failures-artifacts
-                  path: artifacts
```
Comment (alexstine) [inline .github/workflows/end2end-test-playwright.yml:57]: Why does this currently not work? This should support directories by default.

https://github.com/actions/upload-artifact

First location is set here: https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/playwright.config.ts#L11
Second location is set here: https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/playwright.config.ts#L24

Still the same artifacts directory so may be better to figure out why success is not being uploaded.

**Review by alexstine — COMMENTED** [review]
@kevin940726 Any ideas about this?

File: .github/workflows/end2end-test-playwright.yml (hunk 1 above)
Comment (kevin940726) [reply .github/workflows/end2end-test-playwright.yml:57]: We don't want the directory to be uploaded when the tests are successful. The problem is that Playwright will generate the `storage-state` directory inside `artifacts` too, no matter if the tests have flaky results or not. But that's not useful for debugging when there are no flaky tests. So the solution is to only upload the `test-results` directory, which will only exist if there are failed/flaky results.

Comment (alexstine) [reply .github/workflows/end2end-test-playwright.yml:57]: Makes sense now. 👍

**Review by alexstine — APPROVED** [review]
LGTM.

**Pull request merged**